### PR TITLE
Rename `[Simple|Aggregated]MetricSample` to `[Simple|Aggregated]MetricValue`

### LINKED ISSUE #141: Inconsistent time stamp usage in sample definitions
### What's needed?

A sample of a time series is defined by a timestamp and a value (see https://github.com/frequenz-floss/frequenz-sdk-python/blob/v1.x.x/docs/user-guide/glossary.md#sample). The current implementations of `SimpleMetricSample`,  `AggregatedMetricSample` and `MetricSampleVariant` do not contain the timestamp, but `MetricSample` does. 

### Proposed solution

Propose to add a timestamp to the base sample types and remove it from `MetricSample` since it uses the others (would also affect coupling to bounds though).

### Use cases

Base sample types like `SimpleMetricSample` and `MetricSample` are both intended to be used by other APIs like reporting API.

### Alternatives and workarounds

Rename Sample types without timestamp to `Value`.

### Additional context

See discussion in https://github.com/frequenz-floss/frequenz-api-reporting/pull/6/files#r1406773110

### PR DESCRIPTION
**Pull request by tiyash-basu-frequenz**
Closes #141

### CODE REVIEW

**Review by cwasicki — DISMISSED** [review]

File: proto/frequenz/api/common/v1/metrics/metric_sample.proto [hunk 1]
```diff
@@ -46,19 +46,19 @@ message AggregatedMetricSample {
...
 // This message is designed to offer flexibility in capturing different
 // granularities of metric samples—either a simple single-point measurement
 // or an aggregated set of measurements for a metric.
 //
-// A `MetricSampleVariant` can hold either a `SimpleMetricSample` or an
-// `AggregatedMetricSample`, but not both simultaneously. Setting one will
+// A `MetricSampleVariant` can hold either a `SimpleMetricValue` or an
+// `AggregatedMetricValue`, but not both simultaneously. Setting one will
 // nullify the other.
 message MetricSampleVariant {
```
Comment (nhatcher-frequenz) [inline proto/frequenz/api/common/v1/metrics/metric_sample.proto]: This needs to be renamed to `MetricValueVariant` and `metric_sample_type` to `metric_value_type`

**Review by nhatcher-frequenz — CHANGES_REQUESTED** [review]
Minor issue

File: proto/frequenz/api/common/v1/metrics/metric_sample.proto (hunk 1 above)
Comment (tiyash-basu-frequenz) [reply proto/frequenz/api/common/v1/metrics/metric_sample.proto]: Good catch! :bow: 

Fixed

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -30,6 +30,16 @@
...
 
+- Renamed `SimpleMetricSample` to `SimpleMetricValue`, because it does not
+  contain a timestamp, so it does not represent a sample but a value.
+
+- Renamed `AggregatedMetricSample` to `AggregatedMetricValue`, because it does not
+  contain a timestamp, so it does not represent a sample but a value.
+
+- Renamed `MetricSampleVariant` to `MetricValueVariant`.
+
+- Rename `MetricSample.sample` to ``MetricSample.value`.
```
Comment (nhatcher-frequenz) [inline RELEASE_NOTES.md]: Just one back tick.

**Review by nhatcher-frequenz — APPROVED** [review]
LGTM!

**Pull request merged**